src/app.module.ts

```diff
@@ -1,12 +1,35 @@
-import { Module } from '@nestjs/common';
-import { AppController } from './app.controller';
-import { AppService } from './app.service';
-import { RegisterController } from './controllers/register/register.controller';
-import { RegisterService } from './services/register/register.service';
+import { Dependencies, Module } from '@nestjs/common';
+import { DataSource } from 'typeorm';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { User } from './model/user.entity';
+import { RegisterModule } from './modules/register.module';
 
+@Dependencies(DataSource)
 @Module({
-  imports: [],
-  controllers: [AppController, RegisterController],
-  providers: [AppService, RegisterService],
+  imports: [
+    ConfigModule.forRoot({ isGlobal: true }),
+    TypeOrmModule.forRootAsync({
+      imports: [ConfigModule],
+      useFactory: (configService: ConfigService) => ({
+        type: 'postgres',
+        host: configService.get('DB_HOST'),
+        port: +configService.get<number>('DB_PORT'),
+        username: configService.get('DB_USER'),
+        password: configService.get('DB_PASSWORD'),
+        database: configService.get('DB_NAME'),
+        entities: [User],
+        synchronize: true,
```

Comment:
I suggest using a database setup similar to the production. Please read about typerorm cli and migrations. Here are some examples -> 
- https://anjith-p.medium.com/typeorm-database-migrations-in-nestjs-apps-ace923edf1bf
- https://typeorm.io/migrations#creating-a-new-migration
- https://www.oneclickitsolution.com/blog/migrations-with-typeorm/
- https://wanago.io/2022/07/25/api-nestjs-database-migrations-typeorm/